### PR TITLE
fix(middleware-apply-body-checksum): fix request copying with clone

### PR DESCRIPTION
**Pull request by syall**
*Issue #, if available:*

N/A

*Description of changes:*

Fixed a subtle bug when request copying, otherwise methods like `clone()` may not be copied over.

If one or more of the packages in the `/packages` directory has been modified, be sure `yarn changeset add` has been run and its output has
been committed and included in this pull request. See CONTRIBUTING.md.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/middleware-apply-body-checksum/src/applyMd5BodyChecksumMiddleware.ts [hunk 1]
```diff
@@ -30,13 +30,12 @@ export const applyMd5BodyChecksumMiddleware =
           digest = options.streamHasher(options.md5, body);
         }
 
-        request = {
-          ...request,
-          headers: {
-            ...headers,
-            "content-md5": options.base64Encoder(await digest),
-          },
+        const cloned = request.clone();
```
Comment (kuhe) [inline packages/middleware-apply-body-checksum/src/applyMd5BodyChecksumMiddleware.ts:33]: This isn't a safe assumption. 

`HttpRequest.isInstance(request)` is incorrectly implemented. It only checks that a request is duck-like an HttpRequest **interface**. The `clone` method only exists on the `HttpRequest` **class**.

It wasn't a good idea naming interfaces and classes the same, even worse that they aren't identical types.

**Review by kuhe — CHANGES_REQUESTED** [review]
Note to self: implement static `HttpRequest.clone()` that accepts `HttpRequest` interface. Deprecate instance method.

**Review by kuhe — APPROVED** [review]

**Comment by kuhe** [comment]
This is no longer necessary due to https://github.com/smithy-lang/smithy-typescript/pull/1345

**Comment by kuhe** [comment]
re-opening because this PR is still good for code clarity

**Review by hpmellema — APPROVED** [review]

**Pull request merged**